### PR TITLE
☕ New cup templates

### PR DESCRIPTION
**Pull request by killerswin2**
## What type of PR is this.
1. [ ] Bug
2. [x] Change
3. [x] Enhancement

### What have you changed and why?
Information:
    redo of the templates, can build mod to test, need help on that.

### Please specify which Issue this PR Resolves.
closes #XXXX

### Please verify the following and ensure all checks are completed.

1. [ ] Have you loaded the mission in LAN host?
2. [ ] Have you loaded the mission on a dedicated server?

### Is further testing or are further changes required?
1. [ ] No
2. [ ] Yes (Please provide further detail below.)

### How can the changes be tested?
Steps: 

********************************************************
Notes:

### CODE REVIEW

**Review by Lazejun — CHANGES_REQUESTED** [review]
"B_Kitbag_khk" Does not exist in Arma.

>>> Entry: "mineAT" | Invalid classname: CUP_MineE | Classname should be from config CfgMagazines       ------> "CUP_MineE_M"

Bad Casing CUP_optic_pgo7v3 should be "CUP_optic_PGO7V3"
Bad case on classname: CUP_optic_eotech553_black, expected: CUP_optic_Eotech553_Black
Bad case on classname: CUP_optic_compm2_low, expected: CUP_optic_CompM2_low
Bad case on classname: CUP_optic_elcan_specterdr_black, expected: CUP_optic_Elcan_SpecterDR_black

There is a lot more, all can be found in the RPTs

**Review by Lazejun — COMMENTED** [review]
Fixed MissileATlauncher as that was never given to any Loadout,
Moved M47 to MissileATLaunchers as its Guided AT,
Fixed the lightATLaunchers and ATLauncher confusion ----> lightATLaunchers  is single use.
CDF:
Removed 75 round Magazines from non MG Loadouts/Weapons as that will just overfill loadouts resulting in 0 Ammo.
Added the same Vanilla item filtering as RHS has.
Fixed Weapon Categories for AT/AA Launchers,
Swaped Vanilla NVGs for a CUP Variant
Added Smaw with Spotting rounds
Added CDF and AFRF Winter, all Vehicle that are not availible in White/Winter will be the same as Temperate.

File: A3A/addons/core/Templates/fn_detector.sqf [hunk 1]
```diff
@@ -23,6 +23,9 @@ A3A_hasADV = false;
 A3A_hasD3S = false;
 A3A_hasRDS = false;
 A3A_hasKAT = false;
+A3A_hasCUP = false;
+A3A_hasBWMOD = false;
```
Comment (Bob-Murphy) [inline A3A/addons/core/Templates/fn_detector.sqf:27]: remove CUP unrelated content

File: A3A/addons/core/Templates/fn_detector.sqf [hunk 2]
```diff
@@ -23,6 +23,9 @@ A3A_hasADV = false;
...
+A3A_hasTornado = false;
```
Comment (Bob-Murphy) [inline A3A/addons/core/Templates/fn_detector.sqf:28]: remove CUP unrelated content

File: A3A/addons/core/Templates/fn_detector.sqf [hunk 3]
```diff
@@ -56,6 +59,21 @@ if (allowDLCVN && isClass (configFile >> "CfgPatches" >> "vn_weapons")) then {A3
...
+  isClass (configFile >> "cfgPatches" >> "CUP_BaseConfigs") &&                   // cup weapons
+  isClass (configFile >> "cfgPatches" >> "CUP_AirVehicles_Core")                 // cup vehicles
+) then {A3A_hasCUP = true; Info("CUP Detected.") };
+
+// BWMOD Detection
+if ( isClass (configfile >> "CfgPatches" >> "bwa3_common")) then { A3A_hasBWMOD = true; Info("BWMOD.") };
+
+//ADDONS BELOW 
+
+//Project Tornado Dectection
```
Comment (Bob-Murphy) [inline A3A/addons/core/Templates/fn_detector.sqf:74]: remove CUP unrelated content

File: A3A/addons/core/Templates/fn_detector.sqf [hunk 4]
```diff
@@ -56,6 +59,21 @@ if (allowDLCVN && isClass (configFile >> "CfgPatches" >> "vn_weapons")) then {A3
...
 if (isClass (configfile >> "CfgPatches" >> "ffaa_armas")) then {A3A_hasFFAA = true; Info("FFAA Detected.") };
 
+// CUP Detection
+if (
+  isClass (configFile >> "cfgPatches" >> "CUP_Creatures_People_Civil_Russia") && // cup units
+  isClass (configFile >> "cfgPatches" >> "CUP_BaseConfigs") &&                   // cup weapons
+  isClass (configFile >> "cfgPatches" >> "CUP_AirVehicles_Core")                 // cup vehicles
+) then {A3A_hasCUP = true; Info("CUP Detected.") };
+
+// BWMOD Detection
```
Comment (Bob-Murphy) [inline A3A/addons/core/Templates/fn_detector.sqf:69]: remove CUP unrelated content

**Review by Bob-Murphy — CHANGES_REQUESTED** [review]

File: A3A/addons/core/Templates/Templates/CUP/CUP_AI_ION_Arctic.sqf [hunk 5]
```diff
@@ -0,0 +1,926 @@
...
+["vehiclesCargoTrucks", ["CUP_I_Van_Cargo_ION"]] call _fnc_saveToTemplate;
+["vehiclesAmmoTrucks", ["CUP_I_nM1038_Ammo_ION_WIN"]] call _fnc_saveToTemplate;
+["vehiclesRepairTrucks", ["CUP_I_nM1038_Repair_ION_WIN"]] call _fnc_saveToTemplate;
+["vehiclesFuelTrucks", ["CUP_B_MTVR_Refuel_USA"]] call _fnc_saveToTemplate;
+["vehiclesMedical", ["CUP_I_nM1035_amb_ION_WIN"]] call _fnc_saveToTemplate;
+["vehiclesAPCs", ["CUP_I_BTR80A_ION", "CUP_I_BTR80_ION"]] call _fnc_saveToTemplate;
+["vehiclesTanks", ["CUP_B_Challenger2_NATO"]] call _fnc_saveToTemplate;
+["vehiclesAA", ["CUP_I_Hilux_zu23_TK"]] call _fnc_saveToTemplate;
+["vehiclesLightAPCs", ["CUP_I_RG31E_M2_ION"]] call _fnc_saveToTemplate;			//this line determines light APCs
+["vehiclesIFVs", ["CUP_I_FENNEK_GMG_ION", "CUP_I_MATV_HMG_ION", "CUP_I_FENNEK_HMG_ION", "CUP_I_MATV_GMG_ION"]] call _fnc_saveToTemplate;				//this line determines IFVs
```
Comment (Lazejun) [inline A3A/addons/core/Templates/Templates/CUP/CUP_AI_ION_Arctic.sqf:33]: Uses the unused categories

File: A3A/addons/core/Templates/Templates/CUP/CUP_AI_ION_Arctic.sqf [hunk 6]
```diff
@@ -0,0 +1,926 @@
...
+_loadoutData set ["SMGs", []];
+_loadoutData set ["machineGuns", []];
+_loadoutData set ["marksmanRifles", []];
+_loadoutData set ["sniperRifles", []];
+
+_loadoutData set ["lightATLaunchers", [
+    ["CUP_launch_M72A6", "", "", "", ["CUP_M72A6_M"], [], ""]
+]];
+_loadoutData set ["ATLaunchers", [
+    ["CUP_launch_Mk153Mod0_blk", "", "", "CUP_optic_smaw_scope", ["CUP_SMAW_HEAA_M", "CUP_SMAW_HEDP_M"], [], ""],
```
Comment (Lazejun) [inline A3A/addons/core/Templates/Templates/CUP/CUP_AI_ION_Arctic.sqf:91]: Doesent include the Spotting rounds

File: A3A/addons/core/Templates/Templates/CUP/CUP_AI_ION_Arctic.sqf [hunk 7]
```diff
@@ -0,0 +1,926 @@
...
+    ["CUP_launch_Mk153Mod0_blk", "", "", "", ["CUP_SMAW_HEAA_M", "CUP_SMAW_NE_M"], [], ""]
+    ]];
+    
+
+_loadoutData set ["missileATLaunchers", ["CUP_launch_Javelin", "", "", "", ["CUP_Javelin_M"], [], ""]];
```
Comment (Lazejun) [inline A3A/addons/core/Templates/Templates/CUP/CUP_AI_ION_Arctic.sqf:96]: missileATLaunchers is not even given do anyone

File: A3A/addons/core/Templates/Templates/CUP/CUP_AI_ION_Arctic.sqf [hunk 8]
```diff
@@ -0,0 +1,926 @@
...
+    ["CUP_srifle_Mk12SPR", "", "", "CUP_optic_leupoldmk4", ["CUP_20Rnd_556x45_Stanag", "CUP_20Rnd_556x45_Stanag_Tracer_Green"], [], ""],
+    ["CUP_srifle_Mk12SPR", "", "", "CUP_optic_hensoldtzo_low", ["CUP_20Rnd_556x45_Stanag", "CUP_20Rnd_556x45_Stanag_Tracer_Green"], [], ""]
+]];
+
+_militiaLoadoutData set ["sniperRifles", [
+    ["CUP_srifle_G22_blk", "", "CUP_bipod_Harris_1A2_L_BLK", "CUP_optic_leupoldmk4", ["CUP_5Rnd_762x67_G22"], [], ""]
+]];
+
+_militiaLoadoutData set ["lightATLaunchers", []];
+_militiaLoadoutData set ["ATLaunchers", []];
```
Comment (Lazejun) [inline A3A/addons/core/Templates/Templates/CUP/CUP_AI_ION_Arctic.sqf:407]: Militia gets no AT.

File: A3A/addons/core/Templates/Templates/CUP/CUP_AI_ION_Arctic.sqf (hunk 6 above)
Comment (Lazejun) [reply A3A/addons/core/Templates/Templates/CUP/CUP_AI_ION_Arctic.sqf]: Spotting Rounds are a secondary Magazine, not a Primary.

File: A3A/addons/core/Templates/Templates/CUP/CUP_AI_ION_Arid.sqf [hunk 9]
```diff
@@ -0,0 +1,929 @@
...
+_policeLoadoutData set ["vests", ["V_TacVest_gen_F"]];
+_policeLoadoutData set ["helmets", ["H_Beret_gen_F", "H_MilCap_gen_F"]];
+_policeLoadoutData set ["shotGuns", [
+["CUP_sgun_AA12", "", "", "", ["CUP_20Rnd_B_AA12_Buck_00", "CUP_20Rnd_B_AA12_Slug"], [], ""],
+["CUP_sgun_SPAS12", "", "", "", ["CUP_8Rnd_12Gauge_Pellets_No00_Buck", "CUP_8Rnd_12Gauge_Slug"], [], ""]
+]];
+
+_policeLoadoutData set ["SMGs", [
+["CUP_smg_bizon", "", "", "", ["CUP_64Rnd_9x19_Bizon_M", "CUP_64Rnd_Yellow_Tracer_9x19_Bizon_M"], [], ""],
+["CUP_smg_BallisticShield_MP7", "", "", "", ["CUP_40Rnd_46x30_MP7", "CUP_40Rnd_46x30_MP7_Green_Tracer"], [], ""]
```
Comment (Lazejun) [inline A3A/addons/core/Templates/Templates/CUP/CUP_AI_ION_Arid.sqf:349]: Not very useful to give the only Shield that can crash your Game

File: A3A/addons/core/Templates/Templates/CUP/CUP_AI_ION_Arid.sqf [hunk 10]
```diff
@@ -0,0 +1,929 @@
...
+    ["vests"] call _fnc_setVest;
+    ["uniforms"] call _fnc_setUniform;
+    [["atBackpacks", "backpacks"] call _fnc_fallback] call _fnc_setBackpack;
+
+    ["rifles"] call _fnc_setPrimary;
+    ["primary", 5] call _fnc_addMagazines;
+
+    [selectRandom ["missileATLaunchers", "ATLaunchers"]] call _fnc_setLauncher;
+    //TODO - Add a check if it's disposable.
+    ["launcher", 2] call _fnc_addMagazines;
```
Comment (Lazejun) [inline A3A/addons/core/Templates/Templates/CUP/CUP_AI_ION_Arid.sqf:646]: https://github.com/killerswin2/A3-Antistasi/blob/cd8e9647833f61ed6d2ef1273dcebb100c22d822/A3A/addons/core/Templates/Templates/CUP/CUP_AI_US_Marine_Arid.sqf#L633

Comment (killerswin2) [reply A3A/addons/core/Templates/Templates/CUP/CUP_AI_ION_Arid.sqf:645]: resolved in [a1afc4f](https://github.com/official-antistasi-community/A3-Antistasi/pull/2239/commits/a1afc4fc934e46385b37b2d7f8aa96ed2054652b)

File: A3A/addons/core/Templates/Templates/CUP/CUP_AI_ION_Arid.sqf (hunk 9 above)
Comment (killerswin2) [reply A3A/addons/core/Templates/Templates/CUP/CUP_AI_ION_Arid.sqf]: resolved in [a1afc4f](https://github.com/official-antistasi-community/A3-Antistasi/pull/2239/commits/a1afc4fc934e46385b37b2d7f8aa96ed2054652b)

**Comment by killerswin2** [comment]
update cup templates to new ace style. ( for me to remember )

**Comment by killerswin2** [comment]
update tk -> remove mc
update SLA -> remove mc
update RACS -> remove mc
update AFRF -> remove mc

**Review by Lazejun — APPROVED** [review]
Just approving so Merge is not blocked,

This should be seen as WIP,

It may change depending on Community Feedback etc.

A full review will follow at some point.

**Review by Bob-Murphy — APPROVED** [review]
what laze said

**Pull request merged**